Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs

```diff
@@ -0,0 +1,191 @@
+                {
+                    Show();
+                }
+            }
+        }
+
+        static bool AnyTouchDown()
+        {
+            foreach (var touch in Input.touches)
+            {
+                if (touch.phase == TouchPhase.Began)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public void HostIPRequest(string ip, string port)
+        {
+            int.TryParse(port, out var portNum);
+            if (portNum <= 0)
+            {
+                portNum = k_DefaultPort;
+            }
+
+            ip = string.IsNullOrEmpty(ip) ? k_DefaultIP : ip;
+
+            m_GameNetPortal.PlayerName = m_PlayerNameLabel.text;
+
+            m_GameNetPortal.StartHost(ip, portNum);
+
+            m_SignInSpinner.SetActive(true);
+        }
+
+        public void JoinWithIP(string ip, string port)
+        {
+            int.TryParse(port, out var portNum);
+            if (portNum <= 0)
+            {
+                portNum = k_DefaultPort;
+            }
+
+            ip = string.IsNullOrEmpty(ip) ? k_DefaultIP : ip;
+
+            m_GameNetPortal.PlayerName = m_PlayerNameLabel.text;
+
+            m_ClientNetPortal.StartClient(ip, portNum);
+
+            m_SignInSpinner.SetActive(true);
+        }
+
+        public void RegenerateName()
+        {
+            m_PlayerNameLabel.text = m_NameGenerationData.GenerateName();
+        }
+
+        /// <summary>
+        /// Callback when the server sends us back a connection finished event.
+        /// </summary>
+        /// <param name="status"></param>
+        void OnConnectFinished(ConnectStatus status)
+        {
+            // TODO: publish ConnectStatus message
+        }
+
+        public void ToggleJoinIPUI()
+        {
+            m_IPJoiningUI.Show();
+            m_IPHostingUI.Hide();
+            m_JoinTabButtonTinter.SetToColor(1);
+            m_HostTabButtonTinter.SetToColor(0);
+        }
+
+        public void ToggleCreateIPUI()
+        {
+            m_IPJoiningUI.Hide();
+            m_IPHostingUI.Show();
+            m_JoinTabButtonTinter.SetToColor(0);
+            m_HostTabButtonTinter.SetToColor(1);
+        }
+
+        public void Show()
+        {
+            m_CanvasGroup.alpha = 1f;
+            m_CanvasGroup.interactable = true;
+            m_CanvasGroup.blocksRaycasts = true;
+
+            m_SignInSpinner.SetActive(false);
+        }
+
+        public void Hide()
+        {
+            m_CanvasGroup.alpha = 0f;
+            m_CanvasGroup.interactable = false;
+            m_CanvasGroup.blocksRaycasts = false;
+        }
+
+        /// <summary>
+        /// Sanitize user port InputField box allowing only alphanumerics and '.'
+        /// </summary>
+        /// <param name="dirtyString"> string to sanitize. </param>
+        /// <returns> Sanitized text string. </returns>
+        public static string Sanitize(string dirtyString)
```

Comment:
don't we already have sanitization logic somewhere else?

Reply:
Yes so that was moved to the new Lobby scripts from PopupPanel IIRC. So I'd just remove that from the lobby scripts and keep that here.

Reply:
let me know when this is done and I'll approve

Reply:
Good to test now.